### DISCUSSION TITLE
Homeassistant integration with heater control

### DISCUSSION THREAD

**Discussion by Tegguy** [tagyoureit/nodejs-poolController#421] [Q&A]
I should mention I am very new to homeassistant and doing my best to learn

I am tryin to get my homeassistant configuration up and running but have ran into an issue with the heater control.

I have a heatpump with a cool mode built in and as such I have a heatsetPoint and a coolsetPoint. How can I control both of these with the temperature state topic?

I’m also having issues getting the button to change from ultratemp to off but I’m not sure what the command_template is doing.

I've been trying to follow some examples online but am struggling becuse they don't seem to match my setup. For example I have the topic intellicenter-i8ps/state/body/heatMode but both heatmode and id are at 1 rather than one saying off/ultratemp

off: val 1, name off, desc Off
ultratemp: val 5, name ultratemp, desc, UltraTemp

`climate:
  - platform: mqtt
    name: "Pool Heatpump"
    min_temp: 40
    max_temp: 104
    modes:
      - "off"
      - "ultratemp"
    current_temperature_topic: "intellicenter-i8ps/state/temps/bodies/1/pool/temp"
    value_template: "{{ value_json.temp }}"
    mode_state_topic: "intellicenter-i8ps/state/temps/bodies/1/pool/heatMode"
    mode_state_template: >-
      {% if value_json.heatMode.val == 0 %}
        off
      {% elif value_json.heatMode.val == 1 %}
        off
      {% elif value_json.heatMode.val == 5 %}
        ultratemp
      {% endif %}
    mode_command_topic: "intellicenter-i8ps/state/body/heatMode"
    mode_command_template: >-
      {% set values = { 'ultratemp':'{"heatMode":5, "id":1}', 'off':'{"heatMode":1, "id":1}'} %}    
      {{ values[value] if value in values.keys() else '{"heatMode":1, "id":1}' }}
    action_topic: "intellicenter-i8ps/state/temps/bodies/1/pool/heatStatus"
    action_template: >-
      {% if value_json.heatStatus.val == 0 %}
        off
      {% elif value_json.heatStatus.val == 4 %}
        heating
      {% elif value_json.heatStatus.val == 8 %}
        cooling
      {% endif %}
    temperature_command_topic: pool/heatsetpoint #REQUIRES AN AUTOMATION - SEE BELOW
    temperature_state_topic: "intellicenter-i8ps/state/temps/bodies/1/pool/heatsetPoint"
    temperature_state_template: "{{ value_json.heatsetPoint }}"`

**Comment by rstrouse** [comment]
UltraTemp throws some wrinkles that I am sure the HASS folks didn't code for.  This is because there isn't a simple on/off status for the heater.  It can be Heating/Cooling/Off.  The mode is another store since it can have Heater, UltraTemp, UltraTemp Pref, and Off depending on whether you have a gas heater and an UltraTemp installed together.

The setPoint topic has been preserved so that we do not break all the other interfaces out there and it is assumed to be the heatSetpoint.  In fact you can use these monikers interchangeably in the API.  A coolSetpoint was added to extend the topics but it is the only way to set the coolSetpoint in the API.

The off heatMode can be either one of two values 0 or 1.  This is because of the way it is reported and commanded in the Pentair OCPs.  You should treat either mode as off.

heatModes in IntelliCenter
1 = Off
2 = Heater (Gas)
3 = Solar
4 = Solar Preference
5 = UltraTemp
6 = UltraTemp Preference
9 = Heat Pump
11 = MasterTemp
25 = Heat Pump Preference

heatStatus in IntelliCenter
0 = Off
1 = Heater (Gas)
2 = Solar
3 = Solar Cooling
4 = Heat Pump Heat
6 = MasterTemp Heat
8 = Heat Pump Cooling

I hope this helps and does not confuse you more.

**Reply by Tegguy** [reply]
Oh I'm beyond confused at this point trying to learn home assistant however, your answer makes sense. Thank you for the information on the modes and status.

I did open a thread on their forums to see if anyone has any insight into it. I suppose this "wrinkle" only affects the heat and cool setpoint and everything else should be ok (I might have to abandon the coolsetpoint).

I'm still trying to figure out what I'm doing wrong in my code with setting the mode (0 or 5)

**Reply by Tegguy** [reply]
Well I got the mode switch working it would appear only "heat" and "off" are acceptable modes and I was trying to use "Ultratemp"... Still working the coolingsetpoint thing

**Reply by Tegguy** [reply]
I just wanted to follow up on this post. I got this working by using the heat_cool option in home assistant. I have gotten the ability to display the heat and cool setpoints along with the current heater status. I haven't attempted to adjust the heater settings yet but that's on the list
